duo-256.nix

```diff
@@ -190,17 +193,70 @@ in
     };
   };
 
-  users.users.root.initialHashedPassword = "";
+  users.users.root.initialPassword = "milkv";
   services.getty.autologinUser = "root";
 
   services.udev.enable = false;
   services.nscd.enable = false;
-  networking.firewall.enable = false;
-  networking.useDHCP = false;
   nix.enable = false;
   system.nssModules = lib.mkForce [ ];
 
-  environment.systemPackages = with pkgs; [ pfetch python311 ];
+  networking = {
+    interfaces.usb0 = {
+      ipv4.addresses = [
+        {
+          address = "192.168.42.1";
+          prefixLength = 24;
+        }
+      ];
+    };
+    useDHCP = false;
+    hostName = "nixos-duo";
+    firewall.enable = false;
+  };
+
+  systemd.services.rndis-device = {
+    enable = true;
+    description = "Set up usb0 as an RNDIS device";
+    unitConfig = {
+      Type = "oneshot";
+    };
+    serviceConfig = {
+      ExecStart = ''
+        /bin/sh -c "${pkgs.coreutils}/bin/echo device > /proc/cviusb/otg_role"
+      '';
+      ExecStartPost = "${pkgs.coreutils}/bin/sleep 1";
+    };
+    before = [ "network-pre.target" ];
+    wants = [ "network-pre.target" ];
+    wantedBy = [ "network.target" ];
+  };
+
+  services.dnsmasq = {
+    enable = true;
+    settings = {
+      interface = "usb0";
+      dhcp-range = [ "192.168.42.2,192.168.42.2,1h"];
+      dhcp-option = [ "3" "6" ];
+    };
+  };
```

Comment:
I think `systemd-networkd` auto comes with a dhcp server?

Reply:
systemd-network was causing oops'es with a null pointer dereference, so I punted back to native Nix networking + dhcp.  it also has the benefit of matching how the vendor buildroot works.